### PR TITLE
Show ownership and license info on project details dialogs

### LINKED ISSUE #1461: Show ownership and licensing info on project details dialogs
We need to show project ownership info (with any parent project history) as well as the CC-BY-SA license. Follow licensing guidelines described in #1261. 

Here's a sample license html:
```<a href="https://wise.berkeley.edu/project/24268" target="_blank">How Can We Recycle Old Tires? Original</a>is licensed under <a href="http://creativecommons.org/licenses/by-sa/4.0/" target="_blank">CC BY-SA</a> by Spongebob Squarepants. License pertains to original content created by the author(s). Authors are responsible for the usage and attribution of any third-party content linked to or included in this work.```

For a derivative project (copy that has been edited):
```<a href="https://wise.berkeley.edu/project/24269" target="_blank">How Can We Recycle Old Tires? New</a> is licensed under <a href="http://creativecommons.org/licenses/by-sa/4.0/" target="_blank">CC BY-SA</a> by Patrick Star and Squidward Tentacles. This work is a derivative of <a href="https://wise.berkeley.edu/project/24268" target="_blank">How Can We Recycle Old Tires? Original</a> by Spongebob Squarepants, used under <a href="http://creativecommons.org/licenses/by-sa/4.0/" target="_blank">CC BY-SA</a>. License pertains to original content created by the author(s). Authors are responsible for the usage and attribution of any third-party content linked to or included in this work.```

Should probably also show in Authoring, VLE, Classroom Monitor.

Could also last updated/edited date.

### PR DESCRIPTION
**Pull request by breity**
To test:
1. Create a new unit using the authoring tool.
2. Go to `/teacher/library/personal`, find the new unit and click on it to open the project details dialog.
3. Verify that license info and CC BY-SA logo is displayed at the bottom and that `license.txt` file opens when clicking on 'View License' link.
4. Share unit with a new teacher and edit the unit as the shared teacher.
5. Go to `/teacher/library/personal` and click on the unit. Verify that both authors are displayed in the license info.
6. Copy the new project and click on the copied project to open project details dialog.
7. Verify that license info is displayed and that updated `license.txt` loads. Make sure the license info states that unit is a copy of the original unit and displays original authors.
8. Make a change to unit using the authoring tool. Go to `/teacher/library/personal` and click on the copied unit. Verify that license info shows the new author and states that unit is a derivative of the original unit. Verify that updated `license.txt` loads.
9. Repeat steps 6-8 but do so with a unit copy that's created by setting up a run with the original unit.
10. Click on an old unit that doesn't have a `license.txt` file. Verify that license info still shows up in the project details dialog but instead of 'View License' link, there is a 'More' link.

Resolves #1461.

### CODE REVIEW

File: src/main/webapp/site/src/app/domain/parentProject.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export class ParentProject {
+  id: number;
+  title: string;
+  authors: any[];
+  uri: String;
+
+  constructor(jsonObject: any = {}) {
+    for (let key of Object.keys(jsonObject)) {
```
Comment (hirokiterashima) [inline src/main/webapp/site/src/app/domain/parentProject.ts]:
```suggestion
    for (const key of Object.keys(jsonObject)) {
```

Comment (hirokiterashima) [inline src/main/webapp/site/src/app/domain/parentProject.ts]:
```suggestion
    for (const key of Object.keys(jsonObject)) {
```

File: src/main/webapp/site/src/app/modules/library/library-project-details/library-project-details.component.ts [hunk 2]
```diff
@@ -17,55 +19,91 @@ export class LibraryProjectDetailsComponent implements OnInit {
...
+        this.isCopy = true;
+      }
+    }
+  }
+
+  getAuthorsString(authors: any[]): string {
+    if (!authors) {
+      return '';
+    }
+    return authors.map(
```
Comment (hirokiterashima) [inline src/main/webapp/site/src/app/modules/library/library-project-details/library-project-details.component.ts]: I think we should condense into three lines:
```
return authors.map((author) => {
      return `${author.firstName} ${author.lastName}`;
    }).join(', ');
```

File: src/main/webapp/site/src/app/modules/library/library-project-details/library-project-details.component.spec.ts [hunk 3]
```diff
@@ -9,12 +9,15 @@ import { NGSSStandards } from "../ngssStandards";
...
+import { translationsFactory } from "../../../app.module";
+import { I18n } from '@ngx-translate/i18n-polyfill';
+import { ParentProject } from "../../../domain/parentProject";
 
-@Component({ selector: 'app-library-project-menu', template: '' })
-export class LibraryProjectMenuStubComponent {
-  @Input()
-  project: Project;
-}
+// @Component({ selector: 'app-library-project-menu', template: '' })
```
Comment (hirokiterashima) [inline src/main/webapp/site/src/app/modules/library/library-project-details/library-project-details.component.spec.ts]: Remove old code?

File: src/main/webapp/site/src/app/modules/library/library-project-details/library-project-details.component.scss [hunk 4]
```diff
@@ -31,14 +37,31 @@
...
+  padding: 12px;
+
+  img {
+    height: 16px;
+    width: auto;
+    margin-bottom: 2px;
+    vertical-align: middle;
+  }
+
+  //a {
```
Comment (hirokiterashima) [inline src/main/webapp/site/src/app/modules/library/library-project-details/library-project-details.component.scss]: Remove old code?

File: src/main/java/org/wise/portal/presentation/web/controllers/ControllerUtil.java [hunk 5]
```diff
@@ -224,11 +224,20 @@ public static JSONObject getProjectJSON(Project project) throws JSONException {
...
     projectJSON.put("dateCreated", project.getDateCreated());
     projectJSON.put("dateArchived", project.getDateDeleted());
-    projectJSON.put("projectThumb", getProjectThumbIconPath(project));
+    projectJSON.put("projectThumb", getProjectPath(project) + PROJECT_THUMB_PATH);
     projectJSON.put("owner", getOwnerJSON(project.getOwner()));
     projectJSON.put("sharedOwners", getProjectSharedOwnersJSON(project));
     projectJSON.put("parentId", project.getParentProjectId());
     projectJSON.put("wiseVersion", project.getWiseVersion());
+    projectJSON.put("uri", projectService.getProjectURI(project));
+    String licensePath = getProjectLocalPath(project) + LICENSE_PATH;
```
Comment (hirokiterashima) [inline src/main/java/org/wise/portal/presentation/web/controllers/ControllerUtil.java]: Put lines 233->240 into a method.
```
    String licensePath = getProjectLocalPath(project) + LICENSE_PATH;
    File licenseFile = new File(licensePath);
    if (licenseFile.isFile()) {
      licensePath = getProjectPath(project) + LICENSE_PATH;
    } else {
      licensePath = "";
    }
    projectJSON.put("license", licensePath);
```
to
```
   projectJSON.put("license", getLicensePath(project));
```

**Comment by breity** [comment]
@hirokiterashima thanks! Updated with your requested changes.

**Comment by breity** [comment]
@geoffreykwan Thanks. Last commit records and maintains copy history for all parent projects and authors based on our discussion.

**Pull request merged**